### ISSUE TITLE
TFE fails to find some files (json) when launched outside the build directory on Linux

### ISSUE THREAD

**Issue by Ettotsu** [luciusDXL/TheForceEngine#483]
It seems like tfe won't find some necessary files when launched outside the build directory, for example in a window manager with `*.desktop` or from a terminal by typing `theforceengine`. `ExternalData/*.json` files seem to be the problematic ones

For example, in-game gun projectiles won't work.

Here are some extracts from the logs (first one might be normal, though the second one is more of an issue):

```
2024-Dec-22 22:18:46 - [Error : HUD] Cannot load font 'AmoNum2.fnt'
2024-Dec-22 22:18:46 - [Error : HUD] Cannot load font 'SuperWp2.fnt'
2024-Dec-22 22:18:46 - [Error : HUD] Cannot load font 'ArmNum2.fnt'
2024-Dec-22 22:18:46 - [Error : HUD] Cannot load font 'HelNum2.fnt'
```
```
2024-Dec-22 22:18:46 - [LOGICS] Parsing logic JSON(s)
2024-Dec-22 22:18:46 - [EXTERNAL_DATA] Loading pickup data
2024-Dec-22 22:18:46 - [EXTERNAL_DATA] Loading projectile data
2024-Dec-22 22:18:46 - [Error : EXTERNAL_DATA] Warning: Projectile data is incomplete. PROJECTILES.JSON may have been altered. Projectiles may not behave as expected.
2024-Dec-22 22:18:46 - [EXTERNAL_DATA] Loading effects data
2024-Dec-22 22:18:46 - [Error : EXTERNAL_DATA] Warning: Effect data is incomplete. EFFECTS.JSON may have been altered. Effects may not behave as expected.
2024-Dec-22 22:18:46 - [MemoryRegion] Allocated new memory block in region 'Landru' - new size is 1 blocks, total size is '4194304'
2024-Dec-22 22:18:46 - [MemoryRegion] Allocated new memory block in region 'Cutscene' - new size is 1 blocks, total size is '8388608'
2024-Dec-22 22:18:51 - [Player] Setting up level 'SECBASE'
2024-Dec-22 22:18:51 - [Error : Dark Forces Game Script] LevelScript.fs (0, 0) : Failed to open script file 'LevelScript.fs'
2024-Dec-22 22:18:51 - [Warning : level_loadGeometry] Could not open 'ZWIMP10L.BM', using 'default.bm' instead.
2024-Dec-22 22:18:51 - [Warning : level_loadGeometry] Could not open 'ZWIMP11.BM', using 'default.bm' instead.
2024-Dec-22 22:18:51 - [Warning : level_loadINF] Hit the end of INF 'SECBASE' before parsing all items: 58/60
```

**Comment by jerethk** [comment]
@Ettotsu hi, have you tried this with the 1.15 release? Is it still happening?

**Comment by Ettotsu** [comment]
> [@Ettotsu](https://github.com/Ettotsu) hi, have you tried this with the 1.15 release? Is it still happening?

Hi, I just recompiled and reinstalled on latest commit of master. Now,  weapon sprite is "default" (which was not the case before if I remember correctly) and projectiles are still not working properly.
Everything works fine when launched directly from build directory though.

**Comment by Ettotsu** [comment]
[@Ettotsu](https://github.com/Ettotsu) hi, have you tried this with the 1.15 release? Is it still happening?

Hi, I just recompiled and reinstalled on latest commit of master. Now, weapon sprite is "default" (which was not the case before if I remember correctly) and projectiles are still not working properly.
Everything works fine when launched directly from build directory though.

Here is a screenshot for reference

![Image](https://github.com/user-attachments/assets/94e177cf-ed7c-4812-af88-fa256e0ed72a)

**Comment by luciusDXL** [comment]
A lot of things in TFE assume you are running from the same directory as the data, so this is a symptom of that. We'll have to go through the json loading code to see how to mitigate the issue.

**Comment by jerethk** [comment]
Yeah, it's not finding the directory which contains the jsons

**Comment by jerethk** [comment]
> > [@Ettotsu](https://github.com/Ettotsu) hi, have you tried this with the 1.15 release? Is it still happening?
> 
> Hi, I just recompiled and reinstalled on latest commit of master. Now, weapon sprite is "default" (which was not the case before if I remember correctly) and projectiles are still not working properly. Everything works fine when launched directly from build directory though.
> 
> Here is a screenshot for reference
> 
> ![Image](https://github.com/user-attachments/assets/94e177cf-ed7c-4812-af88-fa256e0ed72a)

Yeah it's because weapons are now also externalised to Json as well as the projectiles.

When you run from outside the build dir, can the game find mods that are put in the TFE\Mods directory? Or do you face the same issue there?

**Comment by baconicsynergy** [comment]
I can confirm that this also happens to me on Linux

**Comment by jerethk** [comment]
@baconicsynergy @Ettotsu 
Could you please try building this to see if it fixes the problem?

https://github.com/luciusDXL/TheForceEngine/pull/493

**Comment by Ettotsu** [comment]
> [@baconicsynergy](https://github.com/baconicsynergy) [@Ettotsu](https://github.com/Ettotsu) Could you please try building this to see if it fixes the problem?
> 
> [#493](https://github.com/luciusDXL/TheForceEngine/pull/493)

I just tried on another computer and it seems to be working as expected.
I will try on my other computer this weekend, just to make sure. Anyway, thanks!

**Comment by jerethk** [comment]
Excellent, thank you very much -- I've tested it on windows too and everything is ok.

**Comment by luciusDXL** [comment]
This should be fixed with the code in master. There will be a new release soon that includes the fix. :)

**Comment by Ettotsu** [comment]
> > [@baconicsynergy](https://github.com/baconicsynergy) [@Ettotsu](https://github.com/Ettotsu) Could you please try building this to see if it fixes the problem?
> > [#493](https://github.com/luciusDXL/TheForceEngine/pull/493)
> 
> I just tried on a different computer (as I can't access the one I played the game on for now) and it seems to be working as expected. I will try on my other computer this weekend, just to make sure. Anyway, thanks!

Just recompiled on my usual computer and it works like a charm

**Comment by luciusDXL** [comment]
This should be fixed in version 1.21.
